### PR TITLE
Use cluster role aggregation for service catalog RBAC

### PR DESCRIPTION
**Pull request by jboyd01**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1535639

follow on from https://github.com/openshift/origin/pull/17976

### CODE REVIEW

**Comment by enj** [comment]
/hold

SC stuff does not go in bootstrap.  @jboyd01 ping me on IRC, I will explain further.

File: pkg/oc/bootstrap/docker/openshift/rbac.go [hunk 1]
```diff
@@ -1,53 +0,0 @@
...
-const (
-	ClusterServiceBrokerAdminRoleName = "system:openshift:clusterservicebroker-client"
-)
-
-// GetServiceCatalogRBACDelta returns a cluster role with the required rules to bootstrap service catalog
-func GetServiceCatalogRBACDelta() []rbac.ClusterRole {
-	return []rbac.ClusterRole{
-		{
-			ObjectMeta: v1.ObjectMeta{
-				Name: bootstrappolicy.AdminRoleName,
```
Comment (enj) [inline pkg/oc/bootstrap/docker/openshift/rbac.go]: This PR should be a 6 line change delta where you add the appropriate label to each cluster role and give each cluster role a unique name.  For example:

```
{
	ObjectMeta: v1.ObjectMeta{
		Name: "system:openshift:service-catalog:aggregate-to-admin",
		Labels: map[string]string{"rbac.authorization.k8s.io/aggregate-to-admin": "true"},
},
...
```

File: pkg/oc/bootstrap/docker/openshift/servicecatalog.go [hunk 2]
```diff
@@ -39,18 +38,7 @@ func (h *Helper) InstallServiceCatalog(f *clientcmd.Factory, configDir, publicMa
 		return err
 	}
 
-	for _, role := range GetServiceCatalogRBACDelta() {
```
Comment (enj) [inline pkg/oc/bootstrap/docker/openshift/servicecatalog.go]: @jboyd01 leave this as-is (there should be no changes to this file other than the `GetServiceCatalogRBACDelta` to `GetServiceCatalogClusterRoles` name change).

Comment (jboyd01) [reply pkg/oc/bootstrap/docker/openshift/servicecatalog.go]: Thanks, finally got it.  Tests look good.

File: pkg/oc/bootstrap/docker/openshift/servicecatalog.go [hunk 3]
```diff
@@ -39,17 +39,18 @@ func (h *Helper) InstallServiceCatalog(f *clientcmd.Factory, configDir, publicMa
...
-			RemoveExtraPermissions: false,
-			Role: reconciliation.ClusterRoleRuleOwner{ClusterRole: &role},
-			Client: reconciliation.ClusterRoleModifier{
-				Client: kubeClient.Rbac().ClusterRoles(),
-			},
-		}).Run(); err != nil {
-			return errors.NewError("could not reconcile service catalog cluster role %s", role.Name).WithCause(err)
-		}
+	for _, role := range GetServiceCatalogClusterRoles() {
+
```
Comment (enj) [inline pkg/oc/bootstrap/docker/openshift/servicecatalog.go]: nit: unnecessary newline

**Comment by enj** [comment]
/hold cancel

LGTM though you need to run go fmt.

**Comment by jboyd01** [comment]
applied go fmt

**Comment by pmorie** [comment]
Thanks for the fix, @jboyd01

**Pull request merged**